M2/configure.ac

```diff
@@ -740,6 +740,28 @@ AC_COMPILE_IFELSE([AC_LANG_SOURCE([
 	# 1 "foo" 1 3 4
 	template <class T> struct A { };])],AC_MSG_RESULT(no);ISYSTEM=yes,AC_MSG_RESULT(yes))
 
+dnl Check for openmp and fail if it cannot be found.
+dnl This is required for building csdp.
+AC_MSG_CHECKING(whether OpenMP is supported by the compiler)
+if test ! "$OPENMP_LDFLAGS"
+then 
+  AC_LANG(C++)
+  AC_PROG_CXX
+  AC_OPENMP
+  if test "$ac_cv_prog_cxx_openmp" = unsupported
+  then AC_MSG_ERROR([OpenMP does not work, use a compiler that supports OpenMP])
+  AC_SUBST(OPENMP_CXXFLAGS,"$OPENMP_CXXFLAGS")
+  AC_SUBST(OPENMP_LDFLAGS,"")
+  fi
+else
```

Comment:
What's the point of giving the user the option of making us not test whether openmp works?  It's better to always test it.  Also, what's the point of OPENMP_LDFLAGS?  Just let the user use LDFLAGS

Reply:
The problem is that under clang, the macro fails, as far as I can tell.  I don't really want to rewrite the macro.  It certainly fails under apple clang, and I think that even under clang-6, one needs to add `-fopenmp -lomp` for the linker.  Clang also warns if it has flags it doesn't understand on the command line.

To get OpenMP to work under apple clang, one needs to first
  `brew install llvm`
which gives one access to libomp.a. Of course, after installing llvm, one can use a later, probably better, clang, since llvm includes clang-7 (currently).

After installing that, add to CFLAGS or CXXFLAGS or CPPFLAGS (not sure where it really belongs):
`-I/usr/local/opt/libomp/include -Xpreprocessor -fopenmp`
and add to LDFLAGS, LDLIBS, whichever is better:
`-L/usr/local/opt/libomp/lib -lomp`
(and also the `-Xpreprocessor -fopenmp` should also be there for linking, I think. It is for me, as the CXXFLAGS get used on the linker command.

For latest clang, AC_OPENMP realizes it needs -fopenmp, but doesn't realize one needs to still add -lomp.  Actually, I need to check this case again.  I've been trying too many variations...

I'm not sure the best way to get this to go.  I already have enough trouble doing autoconf stuff...!  I guess we can add -lomp to LDFLAGS on the configure line, and the other stuff too. And change the compiler to include the needed flags above.  Is that better?  Then maybe AC_OPENMP will work on all cases.  But then we will effectively be using openmp in our code, which we don't (currently) need.  

I'll try this last solution, and see if it works under the 3 cases: apple clang, latest clang (6 or 7), and gcc-8.

Reply:
Now that I've rewritten the macro for you, you can test for openmp here and ignore OPENMP_LDFLAGS